### ISSUE TITLE
Information about proxy via apache

### ISSUE THREAD

**Issue by btsimonh** [datech/node-red-contrib-amazon-echo#89]
Hi, 
I'm using a Scargill script node-red on OrangPi, and hit the port 80 issue.
The problem was that port 80 was in use by apache2 on the machine....

So, I thought I would document how I got it to operate - maybe this can be added as an additional option in troubleshooting...

**Proxying via an existing apache2 on port 80**

Choose a non conflicting port - I chose 8008 - and set in the echo hub node.
Check you can access http://youraddr:yourport/description.xml
(e.g. http://192.168.1.210:8008/description.xml

run
`a2enmod proxy_http`

in your apache conf file (e.g. /etc/apache2/sites-available/000-default.conf), add:
```
  ProxyPreserveHost On
  ProxyPass /api http://localhost:8008/api
  ProxyPassReverse /api http://localhost:8008/api
  ProxyPass /description.xml http://localhost:8008/description.xml
  ProxyPassReverse /description.xml http://localhost:8008/description.xml
```
(replacing 8008 with your port of choice)

run
`service apache2 restart`

Check you can access http://youraddr:80/description.xml

Ask alexa to search for devices.

Note:
The description.xml still contains port 8008.  It may be an idea to optionally have two port configurations, one for the port to listen on, and one to be the port to be contacted on on the local LAN.

**Comment by datech** [comment]
Thanks, @btsimonh.

I've added it to the Troubleshooting page.

**Comment by ministryofsillywalks** [comment]
> 
> 
> Thanks, @btsimonh.
> 
> I've added it to the Troubleshooting page.

FYI the link on the troubleshooting page leads to:
https://github.com/datech/issues/89
instead of this issue:
https://github.com/datech/node-red-contrib-amazon-echo/issues/89

**Comment by ministryofsillywalks** [comment]
however trying to accomplish the same thing with caddy server isn't working for me.
The port forward works fine. calling port 80/description.xml on the host machine forwards me to the port specified in the node and shows the description. 
However my echo dot and my alexa app can't find any devices on my home network (same subnet as the node)
Any way to troubleshoot this further?

**Comment by datech** [comment]
@ministryofsillywalks Thanks for pointing the broken links. Fixed.

Looks like your proxy setup was successful, so you may want to check #104 or #66 for further troubleshooting instructions.

**Comment by ministryofsillywalks** [comment]
> @ministryofsillywalks Thanks for pointing the broken links. Fixed.
> 
> Looks like your proxy setup was successful, so you may want to check #104 or #66 for further troubleshooting instructions.

Just saw this morning that then node had been updated to 1.10 with the new option to toggle discovery. Totally missed that. Enabled discovery and now its working fine!
Thanks!

**Comment by dinki** [comment]
> > @ministryofsillywalks Thanks for pointing the broken links. Fixed.
> > Looks like your proxy setup was successful, so you may want to check #104 or #66 for further troubleshooting instructions.
> 
> Just saw this morning that then node had been updated to 1.10 with the new option to toggle discovery. Totally missed that. Enabled discovery and now its working fine!
> Thanks!

Sorry to bring this old topic back up but I am having the exact same problem.  I'm using Caddy with Node Red running in docker as root.  I'm running the echo hub on port 8686 and I can successfully pull http://192.168.0.25:80/description.xml  .  I've tried turning device discovery on/off with no luck.  I'm running NR v1.2.2 and node-red-contrib-amazon-echo v1.10 .  I'm trying to discover from Android Alexa app as well as a FireTv 4k Stick.

In full disclosure, I did have the node-red-contrib-alexa-smart-home previously installed, but I have totally removed it from NR as well as clearing all devices it had found and removed the skill from Alexa app.

Can anyone give me some suggestions?
